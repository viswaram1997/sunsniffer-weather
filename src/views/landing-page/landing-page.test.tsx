import { render,fireEvent } from "@testing-library/react";
import LandingPage from "./index"

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockedUsedNavigate,
}));

describe('Landing Page', () => {

    test('should match snapshot', () => {
        const result = render(<LandingPage />)
        expect(result.baseElement).toMatchSnapshot();
    })

    test("should throw error for input if it's empty", async () => {
        const result = render(<LandingPage />)
        const button = await result.findByTestId('custom-button')
        fireEvent.click(button)
        const label = result.findByText('Please enter the city name')
        expect(label).not.toBeNull()
    })

    test("should update input value", async () => {
        const result = render(<LandingPage />)
        const input:any = await result.findByPlaceholderText('Enter the city name')
        fireEvent.change(input, {target: {value: 'London'}})
        expect(input.value).toEqual('London')
    })
    test("should navigate to weather details page with city name", async () => {
        const result = render(<LandingPage />)
        const input:any = await result.findByPlaceholderText('Enter the city name')
        fireEvent.change(input, {target: {value: 'London'}})
        const button = await result.findByTestId('custom-button')
        fireEvent.click(button)
        expect(mockedUsedNavigate).toBeCalledWith('/London')
    })

});