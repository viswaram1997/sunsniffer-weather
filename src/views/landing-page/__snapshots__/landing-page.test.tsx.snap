// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should match snapshot 1`] = `
<body>
  <div>
    <div
      class="landing-page-wrapper"
    >
      <form
        class="main-content"
        novalidate=""
      >
        <img
          class="brand-logo"
          src="sunsniffer-logo.png"
        />
        <h3
          class="text-center mt-4"
        >
          Weather App
        </h3>
        <div
          class="mb-3 mt-2"
        >
          <label
            class="form-label"
          >
            City Name
          </label>
          <input
            class="form-control"
            name="city-name"
            placeholder="Enter the city name"
            required=""
            type="text"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Please enter the 
            city name
          </div>
        </div>
        <button
          class="btn btn-primary"
          data-testid="custom-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</body>
`;
